code/modules/mob/living/autohiss.dm

```diff
@@ -82,8 +82,44 @@
 		)
 	autohiss_exempt = list(LANGUAGE_VAURCA)
 
+/datum/species/diona
+	autohiss_basic_map = list(
+			"s" = list("ss","sss"),
+			"z" = list("zz","zzz"),
+			"e" = list("ee", "eee")
+		)
+	autohiss_extra_map = list(
+			"a" = list("aa", "aaa"),
+			"i" = list("ii", "iii"),
+			"o" = list("oo", "ooo"),
+			"u" = list("uu", "uuu")
+		)
+	autohiss_exempt = list(
+			LANGUAGE_ROOTSONG
+		)
+
+/datum/species/proc/handle_autoslow(message)

```

Comment:
This probably could just override `/datum/species/proc/handle_autohiss(message, datum/language/lang, mode)` instead of hooking into it.